src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -193,6 +211,47 @@ private void setApplicant1ProceedFlagToYes(CaseData caseData, CaseData.CaseDataB
         }
     }
 
+    private CallbackResponse validateMediationUnavailableDates(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<String> errors = new ArrayList<>();
+        if ((caseData.getApp1MediationAvailability() != null
+            && YES.equals(caseData.getApp1MediationAvailability().getIsMediationUnavailablityExists()))) {
+            checkUnavailable(errors, caseData.getApp1MediationAvailability().getUnavailableDatesForMediation());
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(errors)
+            .build();
+    }
+
+    private void checkUnavailable(List<String> errors,
+                                  List<Element<UnavailableDate>> datesUnavailableList) {
+        if (isEmpty(datesUnavailableList)) {
+            errors.add(UNAVAILABLE_DATE_RANGE_MISSING);
+        } else {
+            for (Element<UnavailableDate> dateRange : datesUnavailableList) {
+                LocalDate dateFrom = dateRange.getValue().getFromDate();
+                LocalDate dateTo = dateRange.getValue().getToDate();
```

Comment:
Here, I propose creating a variable from dateRange.getValue().getUnavailableDateType(). This way, we call the method once, eliminating the need to call it twice within the if section. Additionally, we can encapsulate those if statements within an if-else if structure.

Reply:
Created a variable and changed to else if structure